Source/Core/Core/IOS/ES/Formats.cpp

```diff
@@ -596,20 +597,22 @@ std::vector<std::array<u8, 20>> SharedContentMap::GetHashes() const
 
 std::string SharedContentMap::AddSharedContent(const std::array<u8, 20>& sha1)
 {
-  auto filename = GetFilenameFromSHA1(sha1);
-  if (filename)
-    return *filename;
+  if (auto filename = GetFilenameFromSHA1(sha1))
+    return *std::move(filename);
 
-  const std::string id = fmt::format("{:08x}", m_last_id);
-  Entry entry;
-  std::copy(id.cbegin(), id.cend(), entry.id.begin());
+  Entry& entry = m_entries.emplace_back();
+  // `m_last_id` is a type which cannot exceed 8 characters when formatted in hex.
+  static_assert(sizeof(m_last_id) <= 4);
```

Comment:
```suggestion
  static_assert(sizeof(m_last_id) == 4);
```

If `m_last_id` were to be smaller than 4, that means `entry.id` is smaller than 8, causing `fmt::format_to` to write out of bounds.

Reply:
Oh, whoops.